R/LatentDigraph.R

```diff
@@ -883,3 +901,80 @@ setMethodS3("stronglyConnectedComponent", "LatentDigraph", function(this, node,
 setMethodS3("plot", "LatentDigraph", function(x, ...) {
   plotLatentDigraph(x$L(), x$observedNodes(), x$latentNodes())
 }, appendVarArgs = F)
+
+#' Get the observed parents on a collection of nodes
+#'
+#' @name observedParents
+#' @export observedParents
+#'
+#' @param this the graph object
+#' @param nodes the nodes on which to get the observed parents
+#' @param ... ignored
+observedParents <- function(this, nodes, ...) {
+  UseMethod("observedParents")
+}
+
+#' @rdname   observedParents
+#' @name     observedParents.LatentDigraph
+#' @export
+setMethodS3("observedParents", "LatentDigraph", function(this, nodes, ...) {
+  parents = this$parents(nodes)
+  observedParents <- unique(parents[! parents %in% this$latentNodes()])
+  return(observedParents)
+}, appendVarArgs = F)
+
+
+
+
+#' Get the corresponding mixed graph
+#'
+#' Only works for graphs where the latent nodes are source nodes
+#'
+#' @name getMixedGraph
+#' @export getMixedGraph
+#'
+#' @param this the graph object
+#' @param ... ignored
+getMixedGraph <- function(this, ...) {
+  UseMethod("getMixedGraph")
+}
+
+#' @rdname   getMixedGraph
+#' @name     getMixedGraph.LatentDigraph
+#' @export
+setMethodS3("getMixedGraph", "LatentDigraph", function(this, ...) {
+
+  if (length(this$parents(this$latentNodes())) != 0){
+    stop("Can only procuse mixed graphs for LatentDigraphs where all latent nodes are source nodes.")
```

Comment:
procuse -> procure?

Reply:
It might be even more obvious to write: "In LatentDigraph: mixed graphs are only available when latent nodes have no parents."